\documentclass[12pt]{article}
\usepackage[osf]{garamondx}
\usepackage[garamondx,cmbraces]{newtxmath}
\usepackage{marginnote}
\usepackage[top=2cm, bottom=2cm, outer=3cm, inner=3cm, heightrounded, 
            marginparwidth=2cm, marginparsep=0.5cm]{geometry}
\usepackage{setspace}
\usepackage[backend=biber, style=authoryear, uniquelist=minyear, uniquename=false]{biblatex}
\usepackage[acronym,nonumberlist]{glossaries}
\usepackage{multirow}
\usepackage{cleveref}
\usepackage{array}
\usepackage{float}

\newfloat{data}{hb}{lod}
\floatname{data}{Data}
\crefname{data}{data}{data}

\newcommand{\defn}[1]{\textit{#1}}
\newcommand{\software}[1]{\textit{#1}}

\newcommand{\set}[1]{\left\lbrace#1\right\rbrace}
\newcommand{\sett}[1]{\{#1\}}
\newcommand{\ie}{\textit{i.e.}}
\renewcommand{\emptyset}{\varnothing}
\renewcommand{\vec}[1]{\mathbf{#1}}
\renewcommand{\star}[1]{#1^*}
\renewcommand{\d}{\mathrm{d}\,}

\DeclareMathOperator*{\argmin}{arg\,min}
\DeclareMathOperator*{\argmax}{arg\,max}

\DeclareMathOperator{\Exponential}{Exponential}
\DeclareMathOperator{\Uniform}{Uniform}
\DeclareMathOperator{\E}{E}
\DeclareMathOperator{\Var}{Var}

\DeclareMathOperator{\inc}{in}
\DeclareMathOperator{\out}{out}

\newcommand{\N}{\mathcal{N}}
\renewcommand{\L}{\mathcal{L}}

\newacronym{BA}{BA}{Barab\'asi-Albert}
\newacronym{SI}{SI}{susceptible-infected}
\newacronym{kSVM}{kSVM}{kernel support vector machine}
\newacronym{nLTT}{nLTT}{normalized lineages-through-time}
\newacronym{SMC}{SMC}{sequential Monte-Carlo}
\newacronym{ABC}{ABC}{approximate Bayesian computation}
\newacronym{MSM}{MSM}{men who have sex with men}
\newacronym{HPD}{HPD}{highest posterior density}
\newacronym{GTR}{GTR}{general time-reversible}
\newacronym{MCMC}{MCMC}{Markov chain Monte Carlo}
\newacronym{MAP}{MAP}{maximum \textit{a posteriori}}
\newacronym{IDU}{IDU}{injection drug user}
\newacronym{HIV}{HIV}{human immuno-deficiency virus}

\graphicspath{{../figures/}}
\newcommand{\tablepath}{../tables}
\def\figwidth{0.8\textwidth}

\title{Phylodynamic inference of contact network parameters through approximate
Bayesian computation}
\author{Rosemary M. McCloskey \and Richard H. Liang \and Art F.Y. Poon}

\addbibresource{papers.bib}

\frenchspacing
\onehalfspacing

\begin{document}

<<setup, include=FALSE>>=
    library(netabc)
@

\maketitle

\section*{Background}

When an infectious disease spreads through a population, transmissions are
generally more likely to occur between certain pairs of individuals. Such pairs
must have a particular mode of contact with one another, which varies with the
mode of transmission of the disease. For airborne pathogens, physical proximity
may be sufficient, while for sexually transmitted diseases, sexual or in some
cases blood-to-blood contact is required. The population together with the set
of links between individuals along which transmission can occur is called the
contact network~\autocite{klovdahl1985social, morris1993epidemiology}. The
structure of the contact network underlying an epidemic can profoundly impact
the speed and pattern of the epidemic's expansion. Network structure can
influence the prevalence trajectory~\autocite{o2010contact} and epidemic
threshold~\autocite{barthelemy2005dynamical}, in turn affecting the estimates
of quantities such as effective population
size~\autocite{goodreau2006assessing}. From a public health perspective,
contact networks have been explored as tools for curtailing epidemic spread, by
way of interventions targeted to well-connected
nodes~\autocite{wang2015targeting}. True contact networks are a challenging
type of data to collect, requiring extensive epidemiological
investigation~\autocite{welch2011statistical}.

Viral sequence data, on the other hand, has become relatively inexpensive and
straightforward to collect on a population level. Due to the high mutation rate
of RNA viruses, epidemiological processes impact the course of viral evolution,
thereby shaping the intra-host viral
phylogeny~\autocite{drummond2003measurably}. The term ``phylodynamics'' was
coined to describe this interaction, as well as the growing family of inference
methods to estimate epidemiological parameters from viral
phylogenies~\autocite{grenfell2004unifying}. These methods have revealed
diverse properties of local viral outbreaks, from basic reproductive
number~\autocite{stadler2011estimating}, to the degree of
clustering~\autocite{hughes2009molecular}, to the elevated transmission risk
during acute infection~\autocite{volz2012simple}. On the other hand, although
sophisticated methods have been developed for fitting complex population
genetic models to phylogenies~\autocite{rasmussen2014phylodynamic}, inference
of structural network parameters has to date been limited. However, it has been
shown that network structure has a tangible impact on phylogeny
shape~\autocite{leventhal2012inferring, colijn2014phylogenetic,
goodreau2006assessing, robinson2013dynamics}, suggesting that such statistical
inference might be possible~\autocite{welch2011statistical}.

Survey-based studies of sexual networks~\autocite{liljeros2001web,
schneeberger2004scale, colgate1989risk} have found that these networks have a
degree distribution which follows a power law \autocite[although there has been
some disagreement, see][]{handcock2004likelihood}. That is, number of nodes of
degree $k$ is proportional to $k^-\gamma$ for some constant $\gamma$. These
networks are also referred to as ``scale-free''. One process by which
scale-free networks can be generated is preferential attachment, where nodes
with a high number of contacts attract new connections at an elevated rate. The
first contact network model incorporating preferential attachment was
introduced by \textcite{barabasi1999emergence}, and is now referred to as the
\gls{BA} model. Under this model, networks are formed by iteratively adding
nodes with $m$ new edges each. In the most commonly studied formulation, these
new edges are joined to existing nodes of degree $k$ with probability
proportional to $k$, so that nodes of high degree tend to attract more
connections. \citeauthor{barabasi1999emergence} suggested an extension where
the probability of attaching to a node of degree $k$ is $k^\alpha$ for some
non-negative constant $\alpha$, and we use this extension here.

Previous work offers precedent for the possibility of statistical inference of
structural network parameters. \textcite{britton2002bayesian} develop a
Bayesian approach to estimate the edge density in an Erd\H{o}s-R\'enyi
network~\autocite{erdos1960evolution} given observed infection dates, and
optionally recovery dates. Their approach was later extended by
\textcite{groendyke2011bayesian} and applied to a much larger data set of 188
individuals. \textcite{volz2007susceptible, volz2008sir} developed differential
equations describing the spread of a \gls{SI} epidemic on static and dynamic
contact networks with several degree distributions, which could in principle be
used for inference if observed incidence trajectories were available.
\textcite{brown2011transmission} analysed the degree distribution of an
approximate transmission network, estimated based on genetic similarity and
estimated times of infection, relating 60\% of \gls{HIV}-infected \gls{MSM} in
the United Kingdom. The transmission network is a subgraph of the contact
network which includes only those edges which have already led to a new
infection. The authors found that a Waring distribution, which is produced by a
more sophisticated preferential attachment model, was a good fit to their
estimated network. 

Standard methods of model fitting involve calculation of the likelihood of
observed data under the model. In maximum likelihood estimation, a quantity
proportional to the likelihood is optimized, often through a standard
multi-dimensional numerical optimization procedure. Bayesian methods integrate
prior information by optimizing the posterior probability instead. To avoid
calculation of a normalizing constant, Bayesian inference is often performed
using \gls{MCMC}, which uses likelihood \emph{ratios} in which the normalizing
constants cancel out. Unfortunately, it is generally difficult to explicitly
calculate the likelihood of an observed transmission tree under a contact
network model, even up to a normalizing constant. To do so, it would be
necessary to integrate over all possible networks, and also over all possible
labellings of the internal nodes of the transmission tree. While it is not
known (to us) whether such integration is tractable, a simpler alternative is
offered by likelihood-free methods, namely \gls{ABC}. \gls{ABC} leverages the
fact that, although calculating the likelihood may be impractical, generating
simulated datasets according to a model is often straightforward. If our model
fits the data well, the simulated data it produces should be similar to the
observed data. More formally, if $D$ is the observed data, the posterior
distribution $f(\theta \mid D)$ on model parameters $\theta$ is replaced as the
target of statistical inference by $f(\theta \mid \rho(\hat{D}, D) <
\varepsilon)$, where $\rho$ is a distance function, $\hat{D}$ is a simulated
dataset according to $\theta$, and $\varepsilon$ is a small
tolerance~\autocite{sunnaker2013approximate}. In the specific case when $\rho$
is a kernel function, the approach is known as
kernel-\gls{ABC}~\autocite{nakagome2013kernel, poon2015phylodynamic}.

Here, we apply kernel-\gls{ABC} to the problem of statistical inference of
contact network parameters from an estimated transmission tree, using the tree
kernel developed by \textcite{poon2013mapping}. We then estimate the parameters
of the \gls{BA} model on a variety of simulated and real data sets. Our results
show that the attachment power parameter $\alpha$ can be inferred with
reasonable accuracy, and can vary considerably between epidemics from different
settings.

\section*{Methods}

We implemented a Gillespie simulation algorithm~\autocite{gillespie1976general}
for simulating epidemics and transmission trees over static contact networks,
in the same fashion as several previous studies
\autocite[\textit{e.g.}][]{o2010contact, robinson2013dynamics,
leventhal2012inferring, groendyke2011bayesian, goodreau2006assessing}. To check
that our implementation was correct, we reproduced Figure 1A of
\textcite{leventhal2012inferring} (our \cref{fig:sf1}), which plots the
unbalancedness of transmission trees simulated over four network models at
various levels of pathogen transmissibility. Our program is freely available at
\url{https://github.com/rmcclosk/netabc}.

We chose to study the \gls{BA} network model~\autocite{barabasi1999emergence}.
In addition to $m$ and $\alpha$, we investigated the parameters $N$, which
denotes the total number of nodes in the network, and $I$, which is the number
of infected nodes at which to stop the simulation and sample the transmission
tree. Nodes in our networks followed simple \gls{SI} dynamics, meaning that
they became infected at a rate proportional to their number of infected
neighbours, and never recovered. For all analyses, the transmission trees'
branch lengths were scaled by dividing by their mean. We used the 
\software{igraph} library's implementation of the \gls{BA}
model~\autocite{csardi2006igraph} to generate the graphs. The analyses were run
on Westgrid (\url{https://www.westgrid.ca/}) and a local computer cluster.

\subsection*{Kernel classifiers}

We used the phylogenetic kernel developed by \textcite{poon2013mapping} to test
whether the parameters of the \gls{BA} model had an effect on tree shape. We
simulated 100 networks under each of three different values of $\alpha$: 0.5,
1.0, and 1.5 (300 networks total). The other parameters were fixed to the
following values: $N$ = 5000, $I$ = 1000, and $m$ = 2. A transmission tree with
500 tips was simulated over each network (300 transmission trees total). The
300 trees were compared pairwise with the tree kernel to form a $300 \times
300$ kernel matrix. The kernel meta-parameters $\lambda$ (the ``decay
factor''), and $\sigma$ (the ``radial basis function
variance'')~\autocite[see][]{poon2013mapping}, were set to 0.3 and 4
respectively. We constructed a \gls{kSVM} classifier for $\alpha$ using the
\software{kernlab} package~\autocite{karatzoglou2004kernlab}, and evaluated its
accuracy with 1000 two-fold cross-validations.

Three similar experiments were performed for the other \gls{BA} model
parameters (one experiment per parameter). $m$ was varied between 2, 3, and 4;
$I$ between 500, 1000, and 2000; and $N$ between 3000, 5000, and 8000. The
parameters not being tested were fixed at the values $N$ = 5000, $I$ = 1000,
$m$ = 2, and $\alpha$ = 1. Thus, we performed a total of four kSVM
cross-validations, one for each of the \gls{BA} model parameters $\alpha$, $I$,
$m$, and $N$. We repeated these four cross-validations with different values of
$\lambda$ (0.2, 0.3, and 0.4) and $\sigma$ ($2^{-3}$, $2^{-2}$, \ldots, $2^3$),
as well as on trees with differing numbers of tips (100, 500, and 1000) and in
epidemics of differing size (500, 1000, and 2000). When evaluating the
classifier for $I$, we did not consider trees with 1000 tips, because one of
the tested $I$ values was 500, and the number of tips cannot be larger than
$I$.

For each of the four parameters, we also tested univariate classifier based on
Sackin's index~\autocite{shao1990tree} and an ordinary SVM based on the
\gls{nLTT} statistic~\autocite{janzen2015approximate}.

\subsection*{ABC simulations}

We implemented the adaptive \gls{SMC} algorithm for \gls{ABC} developed by
\textcite{del2012adaptive}. To check that our implementation was correct, we
applied it to the same mixture of Gaussians used by
\citeauthor{del2012adaptive} to demonstrate their method (originally used
by~\textcite{sisson2007sequential}). We were able to obtain a close
approximation to the function (see \cref{fig:smctest}), and attained the
stopping condition used by the authors in a comparable number of steps.

We simulated three transmission trees, each with 500 tips, under every element
of the Cartesian product of these parameter values: $N$ = 5000, $I$ =
\sett{1000, 2000}, $m$ = \sett{2, 3, 4}, and $\alpha$ = \sett{0.0, 0.5, 1,
1.5}. This produced a total of 24 parameter combinations $\times$ three trees
per combination = 72 trees total. The adaptive \gls{ABC} algorithm was applied
to each tree with these priors: $m \sim$ Uniform(1, 5), $\alpha \sim$
Uniform(0, 2), and $(N, I)$ jointly uniform on the region \{$500 \leq N \leq
15000$, $500 \leq I \leq 5000$, $I \leq N$\}. Following
\textcite{del2012adaptive} and \textcite{beaumont2009adaptive}, all proposals
were Gaussian, with variance equal to twice the empirical variance of the
particles. The algorithm was run with 1000 particles, 5 simulated datasets per
particle, and the ``quality'' parameter controlling the decay rate of the
tolerance $\varepsilon$ set to 0.95. We used the same stopping criterion as
\citeauthor{del2012adaptive}, namely when the \gls{MCMC} acceptance rate
dropped below 1.5\%. Point estimates for the parameters were obtained by taking
the highest point of an estimated kernel density on the final set of particles,
using the \software{density} function with the default parameters in
\software{R}. \Gls{HPD} intervals were calculated with the
\software{HPDinterval} function from the \software{R} package
\software{coda}~\autocite{plummer2006coda}.

Two further analyses were performed to address potential sources of error. To
evaluate the effect of model misspecification in the case of heterogeneity
among nodes, we generated a network where half the nodes were attached with
power $\alpha$ = 0.5, and the other half with power $\alpha$ = 1.5. The other
parameters for this network were $N$ = 5000, $I$ = 1000, and $m$ = 2. To
investigate the effects of potential sampling bias, we simulated a transmission
tree where the tips were sampled in a peer-driven fashion, rather than at
random. That is, the probability to sample a node was twice as high if any of
that node's network peers had already been sampled. The parameters of this
network were $N$ = 5000, $I$ = 2000, $m$ = 2, and $\alpha$ = 0.5.

\subsection*{Investigation of published data}

We applied our kernel-ABC method to several published HIV datasets. Because the
\gls{BA} model generates networks with a single connected component, we
specifically searched for datasets which originated from existing clusters,
either phylogenetically or geographically defined. Characteristics of the
datasets we investigated are given in \cref{tab:data}.

\begin{table}[ht]
  \centering
  \input{\tablepath/realdata}
  \caption{Characteristics of published datasets investigated with kernel-ABC.
  Acronyms: MSM, men who have sex with men; IDU, injection drug users.}
  \label{tab:data}
\end{table}

We downloaded all sequences associated with each study from GenBank. For the
\textcite{novitsky2014impact} data, each sequence was aligned pairwise to the
HXB2 reference sequence (Genbank accession number HIVHXB2CG) and the
hypervariable regions were clipped out with \software{BioPython} version
1.66+~\autocite{cock2009biopython}. Sequences were multiply aligned using
\software{MUSCLE} version 3.8.31 \autocite{edgar2004muscle}, and alignments
were manually inspected with \software{Seaview} version 4.4.2
\autocite{gouy2010seaview}. Phylogenies were constructed from the nucleotide
alignments by approximate maximum likelihood using \software{FastTree2} version
2.1.7 with the \gls{GTR} model. Transmission trees were estimated by rooting
and time-scaling the phylogenies by root-to-tip regression, using a modified
version of Path-O-Gen (distributed as part of
BEAST~\autocite{drummond2007beast}) as described
previously~\autocite{poon2015phylodynamic}. 

Two of the datasets \autocite{li2015hiv,novitsky2014impact} were initially much
larger than the others, containing 1265 and 1299 sequences respectively. To
ensure that the analyses were comparable, we reduced these to a number of
sequences similar to the smaller datasets. For the \textcite{li2015hiv} data,
we detected a cluster of size 280 using a patristic distance cutoff of 0.02 as
described previously~\autocite{poon2014impact}. Only sequences within this
cluster were carried forward. For the \textcite{novitsky2014impact} data, no
large clusters were detected using the same cutoff, so we analysed a subtree of
size 180 chosen arbitrarily.

\section*{Results}

\subsection*{Kernel classifiers}

<<classifiers, include=FALSE>>=
    get.kernel.data <- function (param, step) {
        md <- collect.metadata(paste0('../../simulations/kernel-', param, '/', step, '/*'))
        if ('rbf_variance' %in% colnames(md)) {
          md <- subset(md, rbf_variance == 4 & decay_factor == 0.3)
        }
        if (param == 'alpha') {
            md <- subset(md, m == 2)
        }
        d <- setDT(collect.data(rownames(md)))
        if (param %in% c('alpha', 'm')) {
            d[,list(rsquared=mean(rsquared)), by=c('nsimnode', 'ntip')]
        } else if (param == 'I') {
            d[,list(rsquared=mean(rsquared)), by=ntip]
        } else {
            d[,list(rsquared=mean(rsquared)), by=c('I', 'ntip')]
        }
    }
    kernel.alpha <- get.kernel.data('alpha' ,'classifier')
    kernel.m <- get.kernel.data('m', 'classifier')
    kernel.I <- get.kernel.data('I', 'classifier')
    kernel.N <- get.kernel.data('N', 'classifier')

    sackin.alpha <- get.kernel.data('alpha', 'stats-classifier')
    sackin.m <- get.kernel.data('m', 'stats-classifier')
    sackin.I <- get.kernel.data('I', 'stats-classifier')
    sackin.N <- get.kernel.data('N', 'stats-classifier')

    nltt.alpha <- get.kernel.data('alpha', 'nltt-classifier')
    nltt.m <- get.kernel.data('m', 'nltt-classifier')
    nltt.I <- get.kernel.data('I', 'nltt-classifier')
    nltt.N <- get.kernel.data('N', 'nltt-classifier')
@

Accuracy of the \gls{kSVM} classifiers varied based on the parameter being
tested (\cref{fig:rsquared}, left). Classifiers based on two other tree
statistics, the \gls{nLTT} and Sackin's index, generally exhibited worse
performance than the tree kernel, although the magnitude of the disparity
varied between the parameters (\cref{fig:rsquared}, centre and right).
\Cref{fig:rsquared} shows the cross-validation accuracy of each of the tree
classifiers. The results were largely robust to variations in the tree kernel
meta-parameters $\lambda$ and $\sigma$, although accuracy varied between
different epidemic and sampling scenarios
(\cref{fig:alphacrossv,fig:mcrossv,fig:Icrossv,fig:Ncrossv}).

When classifying $\alpha$, the kernel-SVM classifier had an average $R^2$ of 
    \Sexpr{kernel.alpha[,round(mean(rsquared), 2)]},
compared to 
    \Sexpr{nltt.alpha[,round(mean(rsquared), 2)]}
for the \gls{nLTT}-based SVM, and
    \Sexpr{sackin.alpha[,round(mean(rsquared), 2)]}
for the linear regression against Sackin's index. There was little variation
about the mean for different tree and epidemic sizes. No classifier could
accurately identify the $m$ parameter in any epidemic scenario, with average
$R^2$ values of 
  \Sexpr{kernel.m[,round(mean(rsquared), 2)]} for \gls{kSVM},
  \Sexpr{nltt.m[,round(mean(rsquared), 2)]} for the \gls{nLTT}, and
  \Sexpr{sackin.m[,round(mean(rsquared), 2)]}
for Sackin's index. Again, there was little variation in accuracy between
epidemic scenarios, although the accuracy of the \gls{kSVM} was slightly higher
on 1000-tip trees.

The accuracy of classifiers $I$ varied significantly with the number of tips in
the tree. For 100-tip trees, the average $R^2$ values were
  \Sexpr{kernel.I[ntip == 100, round(mean(rsquared), 2)]},
  \Sexpr{nltt.I[ntip == 100, round(mean(rsquared), 2)]}, and
  \Sexpr{sackin.I[ntip == 100, round(mean(rsquared), 2)]}
for the tree kernel, \gls{nLTT}, and Sackin's index respectively. For 500-tip
trees, the values increased to
  \Sexpr{kernel.I[ntip == 500, round(mean(rsquared), 2)]},
  \Sexpr{nltt.I[ntip == 500, round(mean(rsquared), 2)]}, and
  \Sexpr{sackin.I[ntip == 500, round(mean(rsquared), 2)]}.
Finally, the performance of classifiers for $N$ depended heavily on the
epidemic scenario. The $R^2$ of the \gls{kSVM} classifier ranged from
  \Sexpr{kernel.N[,round(min(rsquared), 2)]}
for the smallest epidemic and smallest sample size, to
  \Sexpr{kernel.N[,round(max(rsquared), 2)]}
for the largest. Likewise, $R^2$ for the \gls{nLTT}-based SVM ranged from 
  \Sexpr{nltt.N[,round(min(rsquared), 2)]}
to
  \Sexpr{nltt.N[,round(max(rsquared), 2)]}.
Sackin's index did not accurately classify $N$ in any scenario, with an average
$R^2$ of
  \Sexpr{sackin.N[,round(mean(rsquared), 2)]}
and little variation between scenarios.

\begin{figure}[ht]
  \centering
  \includegraphics[width=\textwidth]{kernel-rsquared.pdf}
  \caption{
      Cross-validation accuracy of kernel-SVM classifier (left), SVM classifier
      using \gls{nLTT} (centre), and linear regression using Sackin's index
      (right) for \gls{BA} model parameters. Kernel meta-parameters were set to
      $\lambda = 0.3$ and $\sigma = 4$. Each point was calculated based on 300
      simulated transmission trees over networks with three different values of
      the parameter being tested. Vertical lines are empirical 95\% confidence
      intervals based on 1000 two-fold cross-validations.
  }
  \label{fig:rsquared}
\end{figure}

\subsection*{ABC simulations}

<<point_est, include=FALSE>>=
    f <- Sys.glob("../../simulations/abc-pa-free-m/point-estimate/*")
    d <- fread(f)
    d[,m := floor(m)]
    d[,alpha_error := abs(true_alpha - alpha)]
    d[,N_error := abs(true_N - N)]
    d[,I_error := abs(true_I - I)]
    d[,m_error := abs(true_m - m)]

    alpha.av <- anova(lm(alpha_error ~ factor(true_alpha) + factor(true_m) + factor(true_I), d))
    m.av <- anova(lm(m_error ~ factor(true_alpha) + factor(true_m) + factor(true_I), d))
    N.av <- anova(lm(N_error ~ factor(true_alpha) + factor(true_m) + factor(true_I), d))
    I.av <- anova(lm(I_error ~ factor(true_alpha) + factor(true_m) + factor(true_I), d))

    m.tbl <- d[,prop.table(table(m_error))]
    zero.alpha.test <- wilcox.test(d[true_alpha == 0, alpha_error], 
                                   d[true_alpha != 0, alpha_error])
    alpha.I.test <- wilcox.test(d[true_alpha < 1, I_error],
                                d[true_alpha >= 1, I_error])
    I.I.test <- wilcox.test(d[true_I == 1000, I_error],
                            d[true_I == 2000, I_error])
    alpha.m.test <- wilcox.test(d[true_alpha == 0 | true_alpha == 1, m_error],
                                d[true_alpha == 0.5 | true_alpha == 1.5, m_error])
    options(scipen=4)
    stopifnot(alpha.av["factor(true_m)", "Pr(>F)"] > 0.05)
    stopifnot(alpha.av["factor(true_I)", "Pr(>F)"] > 0.05)
    stopifnot(I.I.test$p.value < 1e-5)
    stopifnot(I.av["factor(true_m)", "Pr(>F)"] > 0.05)
    stopifnot(m.av["factor(true_m)", "Pr(>F)"] > 0.05)
    stopifnot(m.av["factor(true_I)", "Pr(>F)"] > 0.05)
    stopifnot(min(N.av[1:3,"Pr(>F)"]) > 0.05)
@

\Cref{fig:abcpt} shows \gls{MAP} point estimates of the \gls{BA} model
parameters obtained with kernel-\gls{ABC} on simulated data. The estimates
shown correspond only to the simulations where the $m$ parameter was set to 2,
however the results for $m = 3$ and $m = 4$ were similar 
(\cref{fig:abcptm3,fig:abcptm4}). Average boundaries of 95\% \gls{HPD}
intervals are given in \cref{tab:abchpd}.

The accuracy of the parameter estimates obtained with kernel-\gls{ABC}
paralleled the results from the \gls{kSVM} classifier. Of the four parameters,
$\alpha$ was the most accurately estimated, with point estimates having a
median [IQR] absolute error of 
    \Sexpr{d[,round(median(alpha_error), 2)]} 
    [\Sexpr{d[,round(quantile(alpha_error, 0.25), 2)]} - 
    \Sexpr{d[,round(quantile(alpha_error, 0.75), 2)]}].
The errors when the true value of $\alpha$ was zero were significantly greater
than those for the other values 
    (Wilcoxon rank-sum test, $p$ = \Sexpr{round(zero.alpha.test$p.value, 4)}),
but did not vary across the true values of the other parameters (one-way
ANOVA). Estimates for $I$ were also relatively accurate, with point estimate
errors of
    \Sexpr{d[,round(median(I_error))]} 
    [\Sexpr{d[,round(quantile(I_error, 0.25))]} - 
    \Sexpr{d[,round(quantile(I_error, 0.75))]}].
These errors were significantly higher when the true value of $\alpha$ was
at least 1
    (Wilcoxon rank-sum test, $p$ = \Sexpr{round(alpha.I.test$p.value, 4)})
and when the true value of $I$ was 2000 ($p < 10^{-5}$). The true value of $m$
did not affect the estimates of $I$ (one-way ANOVA).

The $m$ parameter was estimated correctly in only
    \Sexpr{as.integer(m.tbl[1] * 100)} \%
of simulations. Oddly, the error in the estimated $m$ was higher for integer
values of $\alpha$ (\ie~0 and 1) than non-integer values 
    (Wilcoxon rask-sum test, $p$ = \Sexpr{round(alpha.m.test$p.value, 3)}).
The true values of the other parameters did not significantly affect the
estimates of $m$ (both one-way ANOVA). Finally, the total number of nodes $N$
was consistently over-estimated by about a factor of two
    (error \Sexpr{d[,round(median(N_error))]} 
    [\Sexpr{d[,round(quantile(N_error, 0.25))]} - 
     \Sexpr{d[,round(quantile(N_error, 0.75))]}]).
No parameters influenced the accuracy of the $N$ estimates (all one-way ANOVA).

\begin{figure}[ht]
  \centering
  \includegraphics[width=\textwidth]{abc-point-estimate-m2.pdf}
  \caption{
    Point estimates of \gls{BA} model parameters obtained by running kernel-ABC
    on simulated phylogenies without training, for simulations with $m = 2$.
    Dotted lines indicate true values, and limits of the $y$-axes are regions
    of uniform prior density.
  }
  \label{fig:abcpt}
\end{figure}

\begin{table}[ht]
  \centering
  \input{\tablepath/abc-hpd}
  \caption{
    \glsreset{HPD}
    Average \acrlong{MAP} point estimates and 95\% \gls{HPD} interval widths
    for \gls{BA} model parameter estimates obtained with kernel-\gls{ABC}.
    Three transmission trees were simulated under each combination of the
    listed parameter values, and the parameters were estimated with
    kernel-\gls{ABC} without training.
  }
  \label{tab:abchpd}
\end{table}

The dispersion of the \gls{ABC} approximation to the posterior also varied
between the parameters, with narrower \gls{HPD} intervals for the parameters
with more accurate point estimates (\cref{tab:abchpd}). \Cref{fig:abcex} shows
the distributions for for one simulation (equivalent plots for all the
simulations can be found in \cref{dat:abcpost}). \Gls{HPD} intervals around
$\alpha$ and $I$ were narrow relative to the region of nonzero prior density,
whereas the intervals for $m$ and $N$ were more widely dispersed.

\begin{figure}[ht]
  \centering
  \includegraphics[width=\textwidth]{abc-posterior-example.pdf}
  \caption{
    Marginal posterior distributions of \gls{BA} model parameters estimated
    with kernel-\gls{ABC} for a single simulated transmission tree. Dotted
    lines and shaded polygon indicate true values.
  }
  \label{fig:abcex}
\end{figure}

<<mixed_peerdriven, include=FALSE>>=
  library(coda)
  d <- fread("bzcat ../../simulations/abc-pa-mixed-alpha/abc/*")
  d <- d[iter == max(iter)]
  d <- d[sample(1:nrow(d), prob=weight, replace=TRUE)]
  params <- c("alpha", "m", "N", "I")
  setnames(d, c("attach_power", "edges_per_vertex", "nodes", "sim_nodes"),
           params)
  d <- melt(d, measure.vars=params, variable.name="parameter")
  f <- function (x) {
    dens <- density(x)
    hpd <- HPDinterval(mcmc(x))
    list(map=dens$x[which.max(dens$y)], hpd.min=hpd[,"lower"], 
         hpd.max=hpd[,"upper"])
  }
  d <- d[,f(value), by=parameter]
  setkey(d, parameter)

  p <- fread("bzcat ../../simulations/abc-pa-peerdriven/abc/*")
  p <- p[iter == max(iter)]
  p <- p[sample(1:nrow(p), prob=weight, replace=TRUE)]
  params <- c("alpha", "m", "N", "I")
  setnames(p, c("attach_power", "edges_per_vertex", "nodes", "sim_nodes"),
           params)
  p <- melt(p, measure.vars=params, variable.name="parameter")
  f <- function (x) {
    dens <- density(x)
    hpd <- HPDinterval(mcmc(x))
    list(map=dens$x[which.max(dens$y)], hpd.min=hpd[,"lower"], 
         hpd.max=hpd[,"upper"])
  }
  p <- p[,f(value), by=parameter]
  setkey(p, parameter)
@

To test the effect of model misspecification, we simulated one network where
the nodes exhibited heterogeneous preferential attachment power (half 0.5, the
other half 1.5), with $m$ = 2, $N$ = 5000, and $I$ = 1000. The \gls{MAP} [95\%
HPD] estimates for each parameter were: 
$\alpha$, 
  \Sexpr{round(d["alpha", map], 2)} 
  [\Sexpr{round(d["alpha", hpd.min], 2)}-
   \Sexpr{round(d["alpha", hpd.max], 2)}];
$I$,
  \Sexpr{d["I", round(map)]} 
  [\Sexpr{d["I", round(hpd.min)]} -
   \Sexpr{d["I", round(hpd.max)]}];
$m$,
  \Sexpr{d["m", floor(map)]} 
  [\Sexpr{d["m", floor(hpd.min)]} -
   \Sexpr{d["m", floor(hpd.max)]}];
$N$,
  \Sexpr{d["N", round(map)]} 
  [\Sexpr{d["N", round(hpd.min)]}-
   \Sexpr{d["N", round(hpd.max)]}].
To test the effect of sampling bias, we sampled one transmission tree in a
peer-driven fashion, where the probability to sample a node was twice as high
if one of it's peers had already been sampled. The parameters for this
experiment were $N$ = 5000, $m$ = 2, $\alpha$ = 0.5, and $I$ = 2000. The
estimated values were
$\alpha$, 
  \Sexpr{p["alpha", round(map, 2)]} 
  [\Sexpr{p["alpha", round(hpd.min, 2)]} -
   \Sexpr{p["alpha", round(hpd.max, 2)]}];
$I$,
  \Sexpr{p["I", floor(map)]} 
  [\Sexpr{p["I", floor(hpd.min)]} -
   \Sexpr{p["I", floor(hpd.max)]}];
$m$,
  \Sexpr{p["m", floor(map)]} 
  [\Sexpr{p["m", floor(hpd.min)]} -
   \Sexpr{p["m", floor(hpd.max)]}];
$N$,
  \Sexpr{p["N", floor(map)]} 
  [\Sexpr{p["N", floor(hpd.min)]} -
   \Sexpr{p["N", floor(hpd.max)]}].
Both of these results were in line with estimates obtained on other simulated
datasets (\cref{tab:abchpd}).

\subsection*{Real data}

<<realdata, include=FALSE>>=
  d <- fread(Sys.glob("../../simulations/aggregates/hpd/*"))
  map <- dcast.data.table(d, dataset~parameter, value.var="map")
  up <- dcast.data.table(d, dataset~parameter, value.var="hpd.max")
  low <- dcast.data.table(d, dataset~parameter, value.var="hpd.min")
  setkey(map, dataset)
  setkey(up, dataset)
  setkey(low, dataset)
@

There was substantial heterogeneity among the parameter estimates for the five
published HIV datasets we analysed. Two of the
datasets~\autocite{niculescu2015recent, wang2015targeting} had estimated
$\alpha$ values near unity (\gls{MAP} estimate [95\% \gls{HPD}] 
  \Sexpr{map["niculescu2015", round(alpha, 2)]} 
  [\Sexpr{low["niculescu2015", round(alpha, 2)]} - 
   \Sexpr{up["niculescu2015", round(alpha, 2)]}]
and
  \Sexpr{map["wang2015", round(alpha, 2)]} 
  [\Sexpr{low["wang2015", round(alpha, 2)]} -
   \Sexpr{up["wang2015", round(alpha, 2)]}]). 
Another two datasets~\autocite{li2015hiv, cuevas2009hiv} had lower estimated
values and wider \gls{HPD} intervals
  (\Sexpr{map["li2015", round(alpha, 2)]} 
  [\Sexpr{low["li2015", round(alpha, 2)]} - 
  \Sexpr{up["li2015", round(alpha, 2)]}]
and
  \Sexpr{map["cuevas2009", round(alpha, 2)]} 
  [\Sexpr{low["cuevas2009", round(alpha, 2)]} -
   \Sexpr{up["cuevas2009", round(alpha, 2)]}]). 
The \textcite{novitsky2014impact} data had an extremely low estimated $\alpha$
and a very wide \gls{HPD} interval
  (\Sexpr{map["novitsky2014", round(alpha, 2)]} 
  [\Sexpr{low["novitsky2014", round(alpha, 2)]} -
   \Sexpr{up["novitsky2014", round(alpha, 2)]}]). 
For all the datasets except \citeauthor{novitsky2014impact}, estimated values of
$I$ were below 2000, with narrow \gls{HPD} intervals around two of the
datasets
  (\citeauthor{cuevas2009hiv}, \Sexpr{map["cuevas2009", floor(I)]} 
  [\Sexpr{low["cuevas2009", floor(I)]} -
   \Sexpr{up["cuevas2009", floor(I)]}];
   \citeauthor{niculescu2015recent}, \Sexpr{map["niculescu2015", floor(I)]}
  [\Sexpr{low["niculescu2015", floor(I)]} - 
   \Sexpr{up["niculescu2015", floor(I)]}])
and wider intervals around the other two
  (\citeauthor{li2015hiv}, \Sexpr{map["li2015", floor(I)]} 
  [\Sexpr{low["li2015", floor(I)]} -
   \Sexpr{up["li2015", floor(I)]}];
   \citeauthor{wang2015targeting}, \Sexpr{map["wang2015", floor(I)]}
  [\Sexpr{low["wang2015", floor(I)]} - 
   \Sexpr{up["wang2015", floor(I)]}]).
The \citeauthor{novitsky2014impact} data was again the outlier, with a very high
estimated $I$, and \gls{HPD} interval spanning almost the entire prior region
  (\Sexpr{map["novitsky2014", floor(I)]} 
  [\Sexpr{low["novitsky2014", floor(I)]} -
   \Sexpr{up["novitsky2014", floor(I)]}]).
No information was gleaned about the $m$ parameter, with the \gls{HPD} interval
occupying the entire prior region for all datasets. The estimates of $N$ were
similarly uninformative, with the exception that the point estimate for the
\citeauthor{wang2015targeting} data was smaller
  (\Sexpr{map["wang2015", floor(N)]})
than the estimates for other datasets
 (average \Sexpr{map[dataset != "wang2015", floor(mean(N))]}).

\begin{figure}[ht]
  \centering
  \includegraphics{realdata-hpd.pdf}
  \caption{
    \acrlong{MAP} point estimates and 95\% \gls{HPD} intervals for parameters
    of the \gls{BA} network model, fitted to five published \gls{HIV} datasets
    with kernel-\gls{ABC}.
  }
  \label{fig:abchpd}
\end{figure}

\section*{Discussion}

Contact networks can have a strong influence on epidemic progression, and are
potentially useful as a public health tool~\autocite{wang2015targeting,
little2014using}. Despite this, few methods exist for investigating contact
network parameters in a phylodynamic framework~\autocite{groendyke2011bayesian,
volz2008sir, brown2011transmission}. Kernel-ABC is a model-agnostic method
which can be used to investigate any quantity that affects tree shape. In this
work, we developed a kernel-ABC-based method to infer the parameters of a
contact network model. The method is general, and could be applied to any model
from which contact networks can be simulated. We demonstrated the method on the
\gls{BA} model, which is a simple model giving rise to the power law degree
distributions commonly observed in real-world networks. 

By training a kernel-SVM classifier, we found that the $\alpha$ and $I$
parameters, representing preferential attachment power and number of infected
nodes, had a strong influence on tree shape. This was reflected in the relative
accuracy of the kernel-ABC estimates of these parameters. The total number of
nodes $N$ had a weak influence on tree shape, which was most prominent when the
epidemic size $I$ and number of sampled tips were both large. The $m$
parameter, representing the number of edges created in the network per vertex,
did not produce much variation in tree shape, resulting in in both poorly
performing classifiers and uninformative kernel-ABC estimates.

$N$ was almost always significantly over-estimated using kernel-ABC. Since the
prior on $N$ and $I$ is jointly uniform on a triangular region ($I \leq N$),
there is more prior mass on high $N$ values. In retrospect, it is unreasonable
to expect good estimation of $N$, because adding more nodes to a \gls{BA}
network does not change the edge density or overall shape. This can be
illustrated by imagining that we add a small number of nodes to a network after
the epidemic simulation has already been completed. It is possible that none of
these new nodes attains a connection to any infected node. Thus, running the
simulation again on the new, larger network could produce the exact same
transmission tree as before. 

As noted by \textcite{lintusaari2016identifiability}, uniform priors on model
parameters may translate to highly informative priors on quantities of
interest. We observed a non-linear relationship between the preferential
attachment power $\alpha$ and the power law exponent $\gamma$
(\cref{fig:gamma}). Therefore, placing a uniform prior on $\alpha$ between 0
and 2 is equivalent to placing an informative prior that $\gamma$ is close to
2. Therefore, if we were primarily interested in $\gamma$ rather than $\alpha$,
a more sensible choice of prior might have a shape similar to \cref{fig:gamma}
and be bounded above by approximately $\alpha$ = 1.5. This would uniformly
bound $\gamma$ in the region $2 \leq \gamma \leq 4$ commonly reported in the
network literature~\autocite{liljeros2001web, schneeberger2004scale,
colgate1989risk, brown2011transmission}. We note however that
\textcite{jones2003assessment} estimated $\gamma$ values greater than four, in
one case as high as 17, for some datasets, indicating that a wider range of
permitted $\gamma$ values may be warranted.

Our investigation of published HIV datasets indicated heterogeneity in the
contact network structures underlying several distinct local epidemics. The
five datasets analysed fell into three categories (\cref{fig:abchpd}). First,
we estimated a preferential attachment power between 0.5 and 1 for the
epidemics studied by \textcite{cuevas2009hiv} and \textcite{li2015hiv}, with
credible intervals occupying nearly the entire region from 0 to 1.
\citeauthor{cuevas2009hiv} studied a group of newly diagnosed individuals in
the Basque Country, Spain. Although the individuals were of mixed risk groups,
and therefore unlikely to comprise a single contact network, a high proportion
of them (47\%) grouped into local clusters based on genetic distance. The low
estimated attachment power for these data is consistent with the sampled
sequences comprising many distinct sub-networks rather than a single connected
network. \citeauthor{li2015hiv} sampled a large number of acutely infected
\gls{MSM} in Shanghai, China, in which we identified a large cluster from the
phylogeny using a patristic distance cutoff~\autocite{poon2014impact}. The low
attachment power estimated for this dataset was surprising given the high
phylogenetic relatedness of the sequences. It is possible that the number and
diversity of circulating recombinant forms introduced errors into the estimated
viral phylogeny.

For the outbreaks studied by \textcite{niculescu2015recent} and
\textcite{wang2015targeting}, the estimated $\alpha$ was close to one, with a
narrower credible interval than for the other studies.
\citeauthor{niculescu2015recent} studied a recent outbreak among Romanian
\gls{IDU}, while \citeauthor{wang2015targeting} sampled acutely infected
\gls{MSM} in Beijing, China. Both studies discovered a high degree of
phylogenetic relatedness owing to the recent infection times and homogeneous
risk groups of the studied populations. The estimated number of infections for
these datasets were also quite low, although the \gls{HPD} interval for
\citeauthor{wang2015targeting} was much wider than that for
\citeauthor{niculescu2015recent}.

The final studied dataset was an outlier in terms of estimated parameters.
\textcite{novitsky2013phylogenetic} sampled approximately 44\% of the
HIV-infected individuals in the northern area of Mochudi, Botswana. Additional
sampling in a later study~\autocite{novitsky2014impact} brought the genotyping
coverage up to 70\%. Even with such a high sampling coverage, we did not detect
any large clusters using patristic distance, and therefore chose to analyze a
subtree instead. Estimates of $\alpha$ and $N$ both had very wide \gls{HPD}
intervals and were markedly different from the other datasets. The estimated
number of infected nodes was also extremely high, much higher than the
estimated HIV prevalence of the town. Several factors may have contributed to
these results. First, the authors note that the their sample was 75\% female.
In a primarily heterosexual risk environment, removal of a disproportionate
number of males from the network could obfuscate the true network structure,
for example if there were a disproportionate number of highly connected nodes
of one gender. Second, the town in question was in close proximity to the
country's capital, and the authors indicated that a high amount of migration
takes place between the two locations. This suggests that the contact network
may include a much larger group based in the capital city, which would explain
the high estimate of $I$.

When interpreting these results, we caution that the \gls{BA} model is quite
simple and most likely misspecified for these data. In particular, the average
degree of a node in the network is equal to $2m$, and therefore is constrained
to be a multiple of 2. Furthermore, we considered the case $m = 1$, where the
network has no cycles, to be implausible and assigned it zero prior
probability. This forces the average degree to be at least four, which may be
unrealistically high for sexual networks. Additional modelling assumptions
include the network being connected and static, all transmission rates being
equal, no removal after infection, and identical behaviour of all nodes. This
last is particularly problematic, as we showed by simulating a network where
some nodes exhibited a higher attachment power than others. The estimated
attachment power was simply the average of the two values, indicating that,
although we could characterize the network in aggregate, the estimated
parameters could not be said to apply to any individual node. Despite these
issues, we felt it was best to demonstrate the method first on a simple model.
It is possible to fit more complex models which address some of these issues,
such as one incorporating heterogeneous node behaviour. This may prove a
fruitful avenue for future investigations.

Our method has a number of caveats, perhaps the most significant being that it
takes a transmission tree as input. In reality, true transmission trees are not
available and must be approximated, often by way of a viral phylogeny. Although
this has been demonstrated to be a fair
approximation~\autocite[e.g.][]{leitner1996accurate}, and is frequently used in
practice~\autocite[e.g.][]{stadler2013uncovering}, the topologies of a viral
phylogeny and transmission tree can differ
significantly~\autocite{ypma2013relating} due to within-host evolution and the
sampling process. In addition, the \gls{ABC}-\gls{SMC} algorithm is
computationally intensive, taking about a day when run on 20 cores in parallel
with the settings we described in the methods. Nevertheless, our method is
potentially useful to epidemiological researchers interested in the general
characteristics of the network structure underlying disease outbreaks. This
work, and previous work by our group~\autocite{poon2015phylodynamic}, has
demonstrated that kernel-ABC is a broadly applicable and effective framework in
which to perform phylodynamic inference.

\setcounter{figure}{0}
\renewcommand{\thefigure}{S\arabic{figure}}
\setcounter{data}{0}
\renewcommand{\thedata}{S\arabic{data}}

\printbibliography

\newpage
\section*{Supplemental Materials}

\begin{figure}[H]
  \centering
  \includegraphics{leventhal2012fig1.pdf}
  \caption{
      Reproduction of Figure 1A from \textcite{leventhal2012inferring} used to
      check the accuracy of our implementation of Gillespie simulation.
      Transmission trees were simulated over three types of network, with
      pathogen transmissimility varying from 0 to 1. Sackin's index was
      calculated for each simulated transmission tree.
  }
  \label{fig:sf1}
\end{figure}

\begin{figure}[H]
  \centering
  \includegraphics{smc-test.pdf}
  \caption{Approximation of mixture of Gaussians used by
    \textcite{del2012adaptive} and \textcite{sisson2007sequential} to test
    \gls{SMC}. Solid black line indicates true distribution. Grey shaded area
    shows SMC approximation obtained with our implementation, using 10000
    particles with one simulated data point per particle.
  }
  \label{fig:smctest}
\end{figure}

\begin{figure}[H]
  \centering
  \includegraphics{kernel-alpha-crossv.pdf}
  \caption{
    Cross-validation accuracy of kernel-SVM classifiers for $\alpha$ parameter
    of \gls{BA} network model, for various tree kernel meta-parameters and
    epidemic scenarios. Each point was calculated based on 300 simulated
    transmission trees over networks with $\alpha$ = 0.5, 1.0, or 1.5. Dotted
    and and dashed lines indicate, respectively, performance of SVM using the
    \gls{nLTT} statistic, and linear regression using Sackin's index. Facets
    are number of infected nodes before the simulation was stopped ($I$) and
    number of tips in the sampled transmission tree.
  }
  \label{fig:alphacrossv}
\end{figure}

\begin{figure}[H]
  \centering
  \includegraphics{kernel-m-crossv.pdf}
  \caption{Cross-validation accuracy of kernel-SVM classifiers for $m$
      parameter of \gls{BA} network model, for various tree kernel
      meta-parameters and epidemic scenarios. Each point was calculated based
      on 300 simulated transmission trees over networks with $m$ = 2, 3, or 4.
      Dotted and and dashed lines indicate, respectively, performance of SVM
      using the \gls{nLTT} statistic, and linear regression using Sackin's
      index. Facets are number of infected nodes before the simulation was
      stopped ($I$) and number of tips in the sampled transmission tree.
  }
  \label{fig:mcrossv}
\end{figure}

\begin{figure}[H]
  \centering
  \includegraphics{kernel-I-crossv.pdf}
  \caption{Cross-validation accuracy of kernel-SVM classifiers for number of
      infected nodes ($I$) under \gls{BA} network model, for various tree
      kernel meta-parameters and two tree sizes. Each point was calculated
      based on 300 simulated transmission trees over networks with $I$ = 500,
      1000, or 2000. Dotted and and dashed lines indicate, respectively,
      performance of SVM using the \gls{nLTT} statistic, and linear regression
      using Sackin's index. Facets are the number of tips in the sampled
      transmission tree.
  }
  \label{fig:Icrossv}
\end{figure}

\begin{figure}[H]
  \centering
  \includegraphics{kernel-N-crossv.pdf}
  \caption{Cross-validation accuracy of kernel-SVM classifiers for total number
      of nodes ($N$) under \gls{BA} network model, for various tree kernel
      meta-parameters and epidemic scenarios sizes. Each point was calculated
      based on 300 simulated transmission trees over networks with $N$ = 3000,
      5000, or 8000. Dotted and and dashed lines indicate, respectively,
      performance of SVM using the \gls{nLTT} statistic, and linear regression
      using Sackin's index. Facets are the number of tips in the sampled
      transmission tree.
  }
  \label{fig:Ncrossv}
\end{figure}

\begin{figure}[H]
  \centering
  \includegraphics{abc-point-estimate-m3.pdf}
  \caption{
    Point estimates of \gls{BA} model parameters obtained by running kernel-ABC
    on simulated phylogenies without training, for simulations with $m = 3$.
    Dotted lines indicate true values, and limits of the $y$-axes are regions
    of uniform prior density.
  }
  \label{fig:abcptm3}
\end{figure}

\begin{figure}[H]
  \centering
  \includegraphics{abc-point-estimate-m4.pdf}
  \caption{
    Point estimates of \gls{BA} model parameters obtained by running kernel-ABC
    on simulated phylogenies without training, for simulations with $m = 4$.
    Dotted lines indicate true values, and limits of the $y$-axes are regions
    of uniform prior density.
  }
  \label{fig:abcptm4}
\end{figure}

\begin{figure}[H]
  \centering
  \includegraphics{alpha-gamma.pdf}
  \caption{
      Relationship between preferential attachment power parameter $\alpha$
      and power law exponent $\gamma$ for networks simulated under the \gls{BA}
      network model with $N$ = 5000 and $m$ = 2.
  }
  \label{fig:gamma}
\end{figure}

\begin{data}[H]
  \centering
  \includegraphics{stock/missing}
  \caption{
    Plots of marginal posterior distributions estimated with kernel-\gls{ABC}
    for all simulated transmission trees.
  }
  \label{dat:abcpost}
\end{data}

\end{document}
