\documentclass[12pt]{report}
\usepackage[osf]{garamondx}
\usepackage[garamondx,cmbraces]{newtxmath}
\usepackage{marginnote}
\usepackage[top=2cm, bottom=2cm, outer=3cm, inner=3cm, heightrounded, 
            marginparwidth=2cm, marginparsep=0.5cm]{geometry}
\usepackage{setspace}
\usepackage[backend=biber, style=authoryear]{biblatex}
\usepackage{nicefrac}
\usepackage{amsmath}
\usepackage{algorithm}
\usepackage{algorithmicx}
\usepackage{algpseudocode}
\usepackage[acronym,nonumberlist]{glossaries}

\newcommand{\comment}[1]{\marginnote{\textit{#1}}}

\newcommand{\defn}[1]{\textit{#1}}
\newcommand{\software}[1]{\textit{#1}}

\newcommand{\set}[1]{\left\lbrace#1\right\rbrace}
\newcommand{\sett}[1]{\{#1\}}
\renewcommand{\emptyset}{\varnothing}
\renewcommand{\vec}[1]{\mathbf{#1}}
\renewcommand{\star}[1]{#1^*}
\renewcommand{\d}{\mathrm{d}\,}

\DeclareMathOperator*{\argmin}{arg\,min}
\DeclareMathOperator*{\argmax}{arg\,max}

\DeclareMathOperator{\Exponential}{Exponential}
\DeclareMathOperator{\Uniform}{Uniform}
\DeclareMathOperator{\E}{E}
\DeclareMathOperator{\Var}{Var}

\DeclareMathOperator{\inc}{in}
\DeclareMathOperator{\out}{out}

\newcommand{\N}{\mathcal{N}}
\renewcommand{\L}{\mathcal{L}}

\newacronym{BA}{BA}{Barab\'asi-Albert}
\newacronym{SI}{SI}{susceptible-infected}
\newacronym{kSVM}{kSVM}{kernel support vector machine}
\newacronym{nLTT}{nLTT}{normalized lineages-through-time}
\newacronym{SMC}{SMC}{sequential Monte-Carlo}
\newacronym{ABC}{ABC}{approximate Bayesian computation}
\newacronym{MSM}{MSM}{men who have sex with men}

\graphicspath{{figures/}}

\title{Phylodynamic inference of contact network parameters through approximate
Bayesian computation}
\author{Rosemary M. McCloskey \and Richard H. Liang \and Art F.Y. Poon}

\addbibresource{papers.bib}

\frenchspacing
\onehalfspacing

\begin{document}

\maketitle

\section*{Background}

From a theoretical perspective, network structure can profoundly influence the
incidence and prevalence trajectories of an epidemic, in turn affecting the
estimates of quantities such as effective population size. More practically,
contact networks have been explored as tools for curtailing epidemic spread, by
way of interventions targeted to well-connected
nodes~\autocite{little2014using}.

Survey-based studies of sexual networks~\autocite{liljeros2001web,
schneeberger2004scale} have found that sexual contact networks are best
described by a preferential attachment model \autocite[but not all,
see][]{jones2003assessment}. This result has been coroborated by more recent
investigations using phylogenetic methods~\autocite{brown2011transmission}.
Under these models, nodes with a high number of contacts attract new
connections at an elevated rate. Networks produced by preferential attachment
have a power-law degree distribution, meaning that the number of nodes of
degree $k$ is proportional to $k^\gamma$ for some constant $\gamma$. When $2 <
\gamma \leq 3$, the network is referred to as ``scale-free''. The first contact
network model incorporating preferential attachment was introduced by
\textcite{barabasi1999emergence}, and is now referred to as the \gls{BA} model.
Under this model, networks are formed by iteratively adding nodes with $m$ new
edges each. These new edges are joined to existing nodes of degree $k$ with
probability proportional to $k^\alpha$, so that nodes of high degree tend to
attract more connections (in the original paper, only $\alpha = 1$ was
investigated).

Due to their complexity, it is generally difficult to explicitly calculate the
likelihood of an observed transmission tree under a contact network model. To
calculate the normalizing constant, we would need to integrate over all
possible networks, and also over all possible labellings of the internal nodes
of the transmission tree. While it is not known (to us) whether such
integration is tractable, a simpler alternative is offered by likelihood-free
methods, namely \gls{ABC}. \gls{ABC} leverages the fact that, although
calculating the likelihood may be impossible, generating simulated datasets
according to a model is often straightforward. If our model fits the data well,
the simulated data it produces should be similar to the observed data. More
formally, if $D$ is the observed data, the posterior distribution $f(\theta
\mid D)$ on model parameters $\theta$ is replaced as the target of statistical
inference by $f(\theta \mid \rho(\hat{D}, D) < \varepsilon)$, where $\rho$ is a
distance function, $\hat{D}$ is a simulated dataset according to $\theta$, and
$\varepsilon$ is a small tolerance~\autocite{sunnaker2013approximate}. In the
specific case when $\rho$ is a kernel function, the approach is known as
kernel-\gls{ABC}.

Here, we apply kernel-\gls{ABC} to the problem of statistical inference of
contact network parameters from an estimated transmission tree. We then
estimate the parameters of the \gls{BA} model on a variety of simulated and
real data sets. 

\section*{Methods}

We implemented a Gillespie simulation algorithm~\autocite{gillespie1976general}
for simulating epidemics and transmission trees over static contact networks,
as has been done previously \autocite[\textit{e.g.}][]{o2010contact,
robinson2013dynamics, leventhal2012inferring, groendyke2011bayesian}. To check
that our implementation was correct, we reproduced Figure 1 of
\textcite{leventhal2012inferring} (see Figure~\ref{fig:sf1}). 

We chose to study the \gls{BA} network model~\autocite{barabasi1999emergence}.
In addition to $m$ and $\alpha$, we investigated the parameters $N$, which
denotes the total number of nodes in the network, and $I$, which is the number
of infected nodes at which to stop the simulation and sample the transmission
tree. Nodes in our networks followed simple \gls{SI} dynamics, meaning that
they became infected at a rate proportional to their numbers of infected
neighbours, and never recovered. For all analyses, the transmission trees'
branch lengths were scaled by dividing by their mean. We used the 
\software{igraph} library's implementation of the \gls{BA}
model~\autocite{csardi2006igraph} to generate the graphs.

\subsection*{Kernel classifiers}

We used the phylogenetic kernel developed by \textcite{poon2013mapping} to test
whether the parameters of the \gls{BA} model had an effect on tree shape. The
parameters not being tested were fixed at the values $N$ = 5000, $I$ = 1000,
$m$ = 2, and $\alpha$ = 1. The parameters were varied one at a time over the
following values: $N$ = \sett{3000, 5000, 8000}, $I$ = \sett{500, 1000, 2000},
$m$ = \sett{2, 3, 4}, and $\alpha$ = \sett{0.5, 1, 1.5}. For each parameter
set, 100 networks were generated, and a transmission tree was simulated over
each (300 trees per parameter set). The 300 trees were compared pairwise with
the tree kernel to form a $300 \times 300$ kernel matrix. We constructed a
\gls{kSVM} classifier for the parameter of interest using the
\software{kernlab} package~\autocite{karatzoglou2004kernlab}, and evaluated its
accuracy with 1000 two-fold cross-validations. The construction of the kernel
matrices and classifiers was repeated for several combinations of the kernel
meta-parameters $\lambda$ (the ``decay factor''), and $\sigma$ (the ``radial
basis function variance'')~\autocite[see][]{poon2013mapping}, and for
transmission trees of sizes 100, 500, and 1000. We also tested univariate
classifier based on Sackin's index~\autocite{shao1990tree} and an ordinary SVM
based on the \gls{nLTT} statistic~\autocite{janzen2015approximate}.

\subsection*{ABC simulations}

We implemented the adaptive \gls{SMC} algorithm for \gls{ABC} developed by
\textcite{del2012adaptive}. To ensure our implementation was correct, we
applied it to the same mixture of Gaussians used by
\citeauthor{del2012adaptive} to demonstrate their method (originally used
by~\textcite{sisson2007sequential}). We were able to obtain a close
approximation to the function (see Figure~\ref{fig:smctest}), and attained the
stopping condition used by the authors in a comparable number of steps.

We simulated three transmission trees, each with 500 tips, under every element
of the Cartesian product of these parameter values: $N$ = 5000, $I$ =
\sett{1000, 2000}, $m$ = \sett{2, 3, 4}, and $\alpha$ = \sett{0, 0.5, 1, 1.5,
2}. The adaptive \gls{ABC} algorithm was applied to each tree with these
priors: $N$ = Uniform(500, 15000), $I$ = Uniform(500, 5000), $m$ = Uniform(1,
5), and $\alpha$ = Uniform(0, 2). Following \textcite{del2012adaptive} and
\textcite{beaumont2009adaptive}, all proposals were Gaussian, with variance
equal to twice the empirical variance of the particles. Because there must be
at least as many nodes in the network as infected nodes, the proposal ratio was
set to zero whenever $I > N$. The algorithm was run with 1000 particles, 5
simulated datasets per particle, and the ``quality'' parameter set to 0.95. We
use the same stopping criterion as \textcite{del2012adaptive}, namely when the
MCMC acceptance rate dropped below 1.5\%. Point estimates for the parameters
were obtained by taking the highest point of an estimated kernel density on the
final set of particles.

Two further analyses were performed to address potential sources of error. To
evaluate the effect of model misspecification in the case of heterogeneity
among nodes, we generated a network where half the nodes were attached with
power $\alpha$ = 0.5, and the other half with power $\alpha$ = 1.5. The other
parameters for this network were $N$ = 5000, $I$ = 1000, and $m$ = 2. To
investigate the effects of potential sampling bias, we simulated a transmission
tree where the tips were sampled in a peer-driven fashion, rather than at
random. That is, the probability to sample a node was twice as high if any of
that node's network peers had already been sampled. The parameters of this
network were $N$ = 5000, $I$ = 2000, $m$ = 2, and $\alpha$ = 0.5.

\subsection*{Investigation of published data}

We applied our kernel-ABC method to several HIV datasets. Because the \gls{BA}
model generates networks with a single connected component, we specifically
searched for datasets which originated from existing clusters, either
phylogenetically or geographically defined. \textcite{wang2015targeting}
reported on a cohort of acutely infected \gls{MSM} in Beijing, China. Through
phylogenetic analysis, they found that they had ``deeply sampled'' the
local \gls{MSM} network, suggesting that many of the samples could be
epidemiologically related.

We downloaded all sequences associated with each paper from GenBank. Sequences
were multiply aligned using \software{MUSCLE} version 3.8.31, and alignments
were manually inspected with \software{Seaview} version 4.4.2. Phylogenies were
constructed from the nucleotide alignments by approximate maximum likelihood
using \software{FastTree2} version 2.1.7 with the GTR model. Transmission trees
were estimated by rooting and time-scaling the phylogenies by root-to-tip
regression, using a modified version of Path-o-Gen as described
previously~\autocite{poon2015phylodynamic}.

\section*{Results}

\subsection*{Kernel classifiers}

Accuracy of the \gls{kSVM} classifiers varied based on the parameter being
tested. For appropriate choices of the kernel meta-parameters, the classifier
had an $R^2$ above 0.8 for all combinations of tree size and epidemic size. The
$I$ parameter was also classified accurately, with an average $R^2$ of 0.93 for
500-tip trees or 0.70 for 100-tip trees. The $m$ parameter was much harder to
classify, with $R^2$ varying between 0.01 and 0.36 depending on $I$ and the
tree size. Finally, the accuracy of the classifier for $N$ varied widely, from
0.08 with 100 tips and $I = 500$, to 0.82 with 1000 tips and $I = 2000$.
Figure~\ref{fig:crossv} shows the results for $\alpha$; similar figures for the
other parameters can be found in the supplemental materials. Based on
inspection of the cross-validation results, we chose to use the meta-parameters
$\lambda = 0.3$ and $\sigma = 4$ in further analyses, and not to use Sackin's
index or the \gls{nLTT}.

\begin{figure}
  \includegraphics{kernel-alpha-crossv}
  \caption{Cross-validation $R^2$ of kernel-SVM classifier for the $\alpha$
    parameter of the \gls{BA} network model.}
  \label{fig:crossv}
\end{figure}

\section*{Discussion}

The $m$ and $N$ parameters were much harder to recover than $\alpha$ and $I$,
suggesting differing influences of these parameters on tree shape. The $\alpha$
parameter in particular was accurately estimated in most cases, implying that
the strength of preferential attachment has a strong impact on the shape of the
transmission tree.

Our method has a number of caveats, perhaps the most significant being that it
takes a transmission tree as input. In reality, true transmission trees are not
available and must be estimated, often by way of a viral phylogeny. Although
this has been demonstrated to work in practice~\autocite{leitner1996accurate},
the topologies of a viral phylogeny and transmission tree can differ
significantly~\autocite{ypma2013relating}. Additionally, the \gls{BA} model
that we fitted here makes a number of unrealistic modelling assumptions, such
as that the network is connected and static, the nodes obey \gls{SI} dynamics,
and that the underlying behaviour of all nodes is the same. This last is
particularly problematic, as we showed by simulating a network where some nodes
exhibited a higher attachment power than others. The estimated attachment power
was simply the average of the two values, indicating that, although we could
characterize the network in aggregate, the estimated parameters could not be
said to apply to any individual node. Finally, the method is computationally
intensive, taking about a day when run on 20 cores in parallel.

\section*{Conclusions}

We developed a novel method which uses kernel-\gls{ABC} to fit contact network
models to transmission trees, and used it to fit the \gls{BA} preferential
attachment model to both simulated and real data. 

\setcounter{figure}{0}
\renewcommand{\thefigure}{S\arabic{figure}}

\printbibliography

\section*{Supplemental Materials}

\begin{figure}
  \includegraphics{leventhal2012fig1}
  \caption{Reproduction of Figure 1A from \textcite{leventhal2012inferring}.}
  \label{fig:sf1}
\end{figure}

\begin{figure}
  \includegraphics{smc_test}
  \caption{Approximation of mixture of Gaussians used by
    \textcite{del2012adaptive} and \textcite{sisson2007sequential} to test
    \gls{SMC}.}
  \label{fig:smctest}
\end{figure}

\end{document}
