\documentclass[12pt]{article}\usepackage[]{graphicx}\usepackage[]{color}
%% maxwidth is the original width if it is less than linewidth
%% otherwise use linewidth (to make sure the graphics do not exceed the margin)
\makeatletter
\def\maxwidth{ %
  \ifdim\Gin@nat@width>\linewidth
    \linewidth
  \else
    \Gin@nat@width
  \fi
}
\makeatother

\definecolor{fgcolor}{rgb}{0.345, 0.345, 0.345}
\newcommand{\hlnum}[1]{\textcolor[rgb]{0.686,0.059,0.569}{#1}}%
\newcommand{\hlstr}[1]{\textcolor[rgb]{0.192,0.494,0.8}{#1}}%
\newcommand{\hlcom}[1]{\textcolor[rgb]{0.678,0.584,0.686}{\textit{#1}}}%
\newcommand{\hlopt}[1]{\textcolor[rgb]{0,0,0}{#1}}%
\newcommand{\hlstd}[1]{\textcolor[rgb]{0.345,0.345,0.345}{#1}}%
\newcommand{\hlkwa}[1]{\textcolor[rgb]{0.161,0.373,0.58}{\textbf{#1}}}%
\newcommand{\hlkwb}[1]{\textcolor[rgb]{0.69,0.353,0.396}{#1}}%
\newcommand{\hlkwc}[1]{\textcolor[rgb]{0.333,0.667,0.333}{#1}}%
\newcommand{\hlkwd}[1]{\textcolor[rgb]{0.737,0.353,0.396}{\textbf{#1}}}%

\usepackage{framed}
\makeatletter
\newenvironment{kframe}{%
 \def\at@end@of@kframe{}%
 \ifinner\ifhmode%
  \def\at@end@of@kframe{\end{minipage}}%
  \begin{minipage}{\columnwidth}%
 \fi\fi%
 \def\FrameCommand##1{\hskip\@totalleftmargin \hskip-\fboxsep
 \colorbox{shadecolor}{##1}\hskip-\fboxsep
     % There is no \\@totalrightmargin, so:
     \hskip-\linewidth \hskip-\@totalleftmargin \hskip\columnwidth}%
 \MakeFramed {\advance\hsize-\width
   \@totalleftmargin\z@ \linewidth\hsize
   \@setminipage}}%
 {\par\unskip\endMakeFramed%
 \at@end@of@kframe}
\makeatother

\definecolor{shadecolor}{rgb}{.97, .97, .97}
\definecolor{messagecolor}{rgb}{0, 0, 0}
\definecolor{warningcolor}{rgb}{1, 0, 1}
\definecolor{errorcolor}{rgb}{1, 0, 0}
\newenvironment{knitrout}{}{} % an empty environment to be redefined in TeX

\usepackage{alltt}
\usepackage[osf]{garamondx}
\usepackage[garamondx,cmbraces]{newtxmath}
\usepackage[top=2cm, bottom=2cm, outer=3cm, inner=3cm, heightrounded, 
            marginparwidth=2cm, marginparsep=0.5cm]{geometry}
\usepackage{setspace}
\usepackage[backend=biber, style=authoryear, uniquelist=minyear, uniquename=false]{biblatex}
\usepackage[acronym,nonumberlist]{glossaries}
\usepackage{multirow}

\newcommand{\defn}[1]{\textit{#1}}
\newcommand{\software}[1]{\textit{#1}}

\newcommand{\set}[1]{\left\lbrace#1\right\rbrace}
\newcommand{\sett}[1]{\{#1\}}
\renewcommand{\emptyset}{\varnothing}
\renewcommand{\vec}[1]{\mathbf{#1}}
\renewcommand{\star}[1]{#1^*}
\renewcommand{\d}{\mathrm{d}\,}

\DeclareMathOperator*{\argmin}{arg\,min}
\DeclareMathOperator*{\argmax}{arg\,max}

\DeclareMathOperator{\Exponential}{Exponential}
\DeclareMathOperator{\Uniform}{Uniform}
\DeclareMathOperator{\E}{E}
\DeclareMathOperator{\Var}{Var}

\DeclareMathOperator{\inc}{in}
\DeclareMathOperator{\out}{out}

\newcommand{\N}{\mathcal{N}}
\renewcommand{\L}{\mathcal{L}}

\newacronym{BA}{BA}{Barab\'asi-Albert}
\newacronym{SI}{SI}{susceptible-infected}
\newacronym{kSVM}{kSVM}{kernel support vector machine}
\newacronym{nLTT}{nLTT}{normalized lineages-through-time}
\newacronym{SMC}{SMC}{sequential Monte-Carlo}
\newacronym{ABC}{ABC}{approximate Bayesian computation}
\newacronym{MSM}{MSM}{men who have sex with men}
\newacronym{MCMC}{MCMC}{Markov chain Monte Carlo}
\newacronym{GTR}{GTR}{generalized time-reversible}

\graphicspath{{figures/}}

\title{Phylodynamic inference of contact network parameters through approximate
Bayesian computation}
\author{Rosemary M. McCloskey \and Richard H. Liang \and Art F.Y. Poon}

\addbibresource{papers.bib}

\frenchspacing
\onehalfspacing
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
\begin{document}



\maketitle

\section*{Background}

The structure of the contact network underlying a viral epidemic can profoundly
impact the speed and pattern of the epidemic's spread. Network structure can
influence the prevalence trajectory~\autocite{o2010contact} and epidemic
threshold~\autocite{barthelemy2005dynamical}, in turn affecting the estimates
of quantities such as effective population
size~\autocite{goodreau2006assessing}. From a public health perspective,
contact networks have been explored as tools for curtailing epidemic spread, by
way of interventions targeted to well-connected
nodes~\autocite{little2014using}. True contact networks are a challenging type
of data to collect, requiring extensive epidemiological
investigation~\autocite{welch2011statistical}.

Viral sequence data, on the other hand, has become relatively inexpensive and
straightforward to collect on a population level. Due to the high mutation rate
of RNA viruses, epidemiological processes impact the course of viral evolution,
thereby shaping the intra-host viral
phylogeny~\autocite{drummond2003measurably}. The term ``phylodynamics'' was
coined to describe this interaction, as well as the growing family of inference
methods to estimate epidemiological parameters from viral
phylogenies~\autocite{grenfell2004unifying}. These methods have revealed
diverse properties of local viral outbreaks, from basic reproductive
number~\autocite{stadler2011estimating}, to the degree of
clustering~\autocite{hughes2009molecular}, to the elevated transmission risk
during acute infection~\autocite{volz2012simple}. On the other hand, inference
of structural network parameters has to date been limited. However, it has been
shown that network structure has a tangible impact on phylogeny
shape~\autocite{leventhal2012inferring, colijn2014phylogenetic,
goodreau2006assessing, robinson2013dynamics}, suggesting that statistical
inference might be possible~\autocite{welch2011statistical}.

Survey-based studies of sexual networks~\autocite{liljeros2001web,
schneeberger2004scale} have found that sexual contact networks are best
described by a preferential attachment model \autocite[although there has been
some disagreement, see][]{jones2003assessment}. Under these models, nodes with
a high number of contacts attract new connections at an elevated rate. Networks
produced by preferential attachment have a power-law degree distribution,
meaning that the number of nodes of degree $k$ is proportional to $k^\gamma$
for some constant $\gamma$. When $2 < \gamma \leq 3$, the network is referred
to as ``scale-free''. The first contact network model incorporating
preferential attachment was introduced by \textcite{barabasi1999emergence}, and
is now referred to as the \gls{BA} model. Under this model, networks are formed
by iteratively adding nodes with $m$ new edges each. These new edges are joined
to existing nodes of degree $k$ with probability proportional to $k^\alpha$, so
that nodes of high degree tend to attract more connections (in the original
paper, only $\alpha = 1$ was investigated).

Previous work offers precedent for the possibility of statistical inference of
structural network parameters. \textcite{britton2002bayesian} develop a
Bayesian approach to estimate the edge density in an Erd\H{o}s-R\'enyi
network~\autocite{erdos1960evolution} given observed infection dates, and
optionally recovery dates. Their approach was later extended by
\textcite{groendyke2011bayesian} and applied to a much larger data set of 188
individuals. \textcite{brown2011transmission} fit a preferential attachment
model to a phylogenetically estimated transmission network relating 60\% of HIV
infected \gls{MSM} in the United Kingdom. The transmission network is a
subgraph of the contact network which includes only those edges which have
already led to a new infection. 

Due to their complexity, it is generally difficult to explicitly calculate the
likelihood of a transmission tree under a contact network model. To do so, we
would need to integrate over all possible networks generated by that model, and
also over all possible labellings of the internal nodes of the transmission
tree. While it is not known (to us) whether such integration is tractable, a
simpler alternative is offered by likelihood-free methods, namely
\gls{ABC}~\autocite{rubin1984bayesianly, tavare1997inferring}. \gls{ABC}
leverages the fact that, although calculating the likelihood may be impossible,
generating simulated datasets according to a model is often straightforward. If
our model fits the data well, the simulated data it produces should be similar
to the observed data. More formally, if $D$ is the observed data, the posterior
distribution $f(\theta \mid D)$ on model parameters $\theta$ is replaced as the
target of statistical inference by $f(\theta \mid \rho(\hat{D}, D) <
\varepsilon)$, where $\rho$ is a distance function, $\hat{D}$ is a simulated
dataset according to $\theta$, and $\varepsilon$ is a small
tolerance~\autocite{sunnaker2013approximate}. In the specific case when $\rho$
is a kernel function, the approach is known as
kernel-\gls{ABC}~\autocite{nakagome2013kernel}.

Here, we apply kernel-\gls{ABC} to the problem of statistical inference of
contact network parameters from an estimated transmission tree, using the tree
kernel developed by \textcite{poon2013mapping}. We then estimate the parameters
of the \gls{BA} model on a variety of simulated and real data sets.

\section*{Methods}

We implemented a Gillespie simulation algorithm~\autocite{gillespie1976general}
for simulating epidemics and transmission trees over static contact networks,
as has been done previously \autocite[\textit{e.g.}][]{o2010contact,
robinson2013dynamics, leventhal2012inferring, groendyke2011bayesian}. To check
that our implementation was correct, we reproduced Figure 1A of
\textcite{leventhal2012inferring} (see Figure~\ref{fig:sf1}). 

We chose to study the \gls{BA} network model~\autocite{barabasi1999emergence}.
In addition to $m$ and $\alpha$, we investigated the parameters $N$, which
denotes the total number of nodes in the network, and $I$, which is the number
of infected nodes at which to stop the simulation and sample the transmission
tree. Nodes in our networks followed simple \gls{SI} dynamics, meaning that
they became infected at a rate proportional to their numbers of infected
neighbours, and never recovered. For all analyses, the transmission trees'
branch lengths were scaled by dividing by their mean. We used the 
\software{igraph} library's implementation of the \gls{BA}
model~\autocite{csardi2006igraph} to generate the graphs. The analyses were run
on Westgrid (\url{https://www.westgrid.ca/}) and a local computer cluster.

\subsection*{Kernel classifiers}

We used the phylogenetic kernel developed by \textcite{poon2013mapping} to test
whether the parameters of the \gls{BA} model had an effect on tree shape. The
parameters not being tested were fixed at the values $N$ = 5000, $I$ = 1000,
$m$ = 2, and $\alpha$ = 1. The parameters were varied one at a time over the
following values: $N$ = \sett{3000, 5000, 8000}, $I$ = \sett{500, 1000, 2000},
$m$ = \sett{2, 3, 4}, and $\alpha$ = \sett{0.5, 1, 1.5}. For each parameter
set, 100 networks were generated, and a transmission tree was simulated over
each (300 trees per parameter set). The 300 trees were compared pairwise with
the tree kernel to form a $300 \times 300$ kernel matrix. We constructed a
\gls{kSVM} classifier for the parameter of interest using the
\software{kernlab} package~\autocite{karatzoglou2004kernlab}, and evaluated its
accuracy with 1000 two-fold cross-validations. The construction of the kernel
matrices and classifiers was repeated for several combinations of the kernel
meta-parameters $\lambda$ (the ``decay factor''), and $\sigma$ (the ``radial
basis function variance'')~\autocite[see][]{poon2013mapping}, and for
transmission trees of sizes 100, 500, and 1000. We also tested univariate
classifier based on Sackin's index~\autocite{shao1990tree} and an ordinary SVM
based on the \gls{nLTT} statistic~\autocite{janzen2015approximate}.

\subsection*{ABC simulations}

We implemented the adaptive \gls{SMC} algorithm for \gls{ABC} developed by
\textcite{del2012adaptive}. To ensure our implementation was correct, we
applied it to the same mixture of Gaussians used by
\citeauthor{del2012adaptive} to demonstrate their method \autocite[originally
used by][]{sisson2007sequential}. We were able to obtain a close approximation
to the function (see Figure~\ref{fig:smctest}), and attained the stopping
condition used by the authors in a comparable number of steps.

We simulated three transmission trees, each with 500 tips, under every element
of the Cartesian product of these parameter values: $N$ = 5000, $I$ =
\sett{1000, 2000}, $m$ = \sett{2, 3, 4}, and $\alpha$ = \sett{0.0, 0.5, 1, 1.5}.
The adaptive \gls{ABC} algorithm was applied to each tree with these priors:
$m \sim$ Uniform(1, 5), $\alpha \sim$ Uniform(0, 2), and $(N, I)$ jointly
uniform on the triangular region \{$500 \leq N \leq 15000$, $500 \leq I \leq
 5000$, $I \leq N$\}. Following \textcite{del2012adaptive} and
\textcite{beaumont2009adaptive}, all proposals were Gaussian, with variance
equal to twice the empirical variance of the particles. The algorithm was run
with 1000 particles, 5 simulated datasets per particle, and the ``quality''
parameter set to 0.95. We use the same stopping criterion as
\textcite{del2012adaptive}, namely when the MCMC acceptance rate dropped below
1.5\%. Point estimates for the parameters were obtained by taking the highest
point of an estimated kernel density on the final set of particles.

Two further analyses were performed to address potential sources of error. To
evaluate the effect of model misspecification in the case of heterogeneity
among nodes, we generated a network where half the nodes were attached with
power $\alpha$ = 0.5, and the other half with power $\alpha$ = 1.5. The other
parameters for this network were $N$ = 5000, $I$ = 1000, and $m$ = 2. To
investigate the effects of potential sampling bias, we simulated a transmission
tree where the tips were sampled in a peer-driven fashion, rather than at
random. That is, the probability to sample a node was twice as high if any of
that node's network peers had already been sampled. The parameters of this
network were $N$ = 5000, $I$ = 2000, $m$ = 2, and $\alpha$ = 0.5.

\subsection*{Investigation of published data}

We applied our kernel-ABC method to several HIV datasets. Because the \gls{BA}
model generates networks with a single connected component, we specifically
searched for datasets which originated from existing clusters, either
phylogenetically or geographically defined. Characteristics of the datasets we
investigated are given in Table~\ref{tab:data}.

\begin{table}
  \centering
  \begin{tabular}{ccccc}
    Reference & Sequences ($n$) & Location & Risk group & Gene \\
    \hline
    \autocite{wang2015targeting} & 173 & Beijing, China & MSM & \textit{pol} \\
    \autocite{cuevas2009hiv} & 287 & Basque Country, Spain & mixed & \textit{pol} \\
    \autocite{novitsky2013phylogenetic} & \multirow{2}{*}{1299} & \multirow{2}{*}{Mochudi, Botswana} & \multirow{2}{*}{mixed} & \multirow{2}{*}{\textit{env}} \\
    \autocite{novitsky2014impact} \\
    \hline
  \end{tabular}
  \caption{Characteristics of published datasets investigated with kernel-ABC.
  Acronyms: MSM, men who have sex with men.}
  \label{tab:data}
\end{table}

We downloaded all sequences associated with each study from GenBank. Sequences
were multiply aligned using \software{MUSCLE} version 3.8.31
\autocite{edgar2004muscle}, and alignments were manually inspected with
\software{Seaview} version 4.4.2 \autocite{gouy2010seaview}. Phylogenies were
constructed from the nucleotide alignments by approximate maximum likelihood
using \software{FastTree2} version 2.1.7 \autocite{price2010fasttree} with the
\gls{GTR} model. Transmission trees were estimated by rooting and time-scaling
the phylogenies by root-to-tip regression, using a modified version of
Path-o-Gen \autocite[distributed as part of \software{BEAST},
][]{drummond2007beast}, as described
previously~\autocite{poon2015phylodynamic}. 

Two additional steps were undertaken for the
\textcite{novitsky2013phylogenetic, novitsky2014impact} data. First, before
performing the multiple sequence alignment, we aligned each
sequence pairwise to the HXB2 reference sequence (GenBank accession number
HIVHXB2CG) using \software{MAFFT} version 7.130b \autocite{katoh2013mafft} and
clipped out the hypervariable regions with \software{Biopython}
version 1.66+ \autocite{cock2009biopython}. Second, after multiple sequence
alignment, the sequences were trimmed to an informative region with
\software{trimAl} version 1.4.rev15 \autocite{capella2009trimal}.

Kernel-ABC was used to fit the \gls{BA} model to each dataset. To make the
analyses comparable, the same priors and configuration parameters were used for
each dataset. The priors were as follows: $\alpha \sim$ Uniform(0, 2), $m \sim$
Uniform(1, 5), and $(N, I) \sim$ Uniform(\sett{$n \leq N \leq 10000$, $n \leq I
\leq 1000$, $I \leq N$}), where $n$ is the number of sequences in the dataset.
The kernel meta-parameters and SMC configuration were the same as in the
\gls{ABC} simulations.

\section*{Results}

\subsection*{Kernel classifiers}



Accuracy of the \gls{kSVM} classifiers varied based on the parameter being
tested. For appropriate choices of the kernel meta-parameters, the classifier
had an $R^2$ above 
    0.8 
for all combinations of tree size and epidemic size. The $I$ parameter was also
classified accurately, with an average $R^2$ of 
    0.93 
for 500-tip trees or 
    0.7 
for 100-tip trees. The $m$ parameter was much harder to classify, with $R^2$
varying between 
    0.004
and
    0.36
depending on $I$ and the tree size. Finally, the accuracy of the classifier for
$N$ varied widely, from 
    0.08
to
    0.82.
Figure~\ref{fig:crossv} shows the results for $\alpha$; similar figures for the
other parameters can be found in the supplemental materials. Based on
inspection of the cross-validation results, we chose to use the meta-parameters
$\lambda = 0.3$ and $\sigma = 4$ in further analyses, and not to use Sackin's
index or the \gls{nLTT}.

\begin{figure}
  \includegraphics{kernel-alpha-crossv}
  \caption{Cross-validation $R^2$ of kernel-SVM classifier for the $\alpha$
    parameter of the \gls{BA} network model.}
  \label{fig:crossv}
\end{figure}

\subsection*{ABC simulations}

Figure~\ref{fig:abcpt} shows point estimates for the simulations with $m = 2$.
The results for the other values of $m$ are similar and can be found in the
supplemental materials.

The marginal approximated posterior distributions for one simulated dataset are
shown in Figure~\ref{fig:abcex}.

\begin{figure}
    \includegraphics{abc-point-est}
    \caption{Point estimates of \gls{BA} model parameters obtained by running
    kernel-ABC on simulated phylogenies. Dotted lines indicate the true values,
    and limits of the $y$-axes are the regions of uniform prior density.}
    \label{fig:abcpt}
\end{figure}

\begin{figure}
    \includegraphics{abc-example}
    \caption{Marginal estimated posterior distributions obtained with ABC on a
    single simulated dataset. Dotted lines and shaded polygon indicate true
    values.}
    \label{fig:abcex}
\end{figure}

\subsection*{Real data}

\section*{Discussion}

% kernel classifiers

The $m$ and $N$ parameters were much harder to recover than $\alpha$ and $I$,
suggesting differing influences of these parameters on tree shape. The $\alpha$
parameter in particular was accurately estimated in most cases, implying that
the strength of preferential attachment has a strong impact on the shape of the
transmission tree.

% abc simulations

The total number of nodes $N$ was almost always significantly over-estimated.
This suggests that $N$ has very little effect on tree shape, and the estimates
were influenced primarily by the prior. Since the prior on $N$ and $I$ is
jointly uniform on a triangular region, there is more prior mass on high $N$
values. In retrospect, this is reasonable because adding more nodes to a
\gls{BA} network does not change the edge density or the network's overall
shape. This can illustrated by imagining that we grow the network after the
epidemic simulation has already been completed. It is possible (though perhaps
unlikely) that none of the new nodes attains a connection to any infected node.
Thus, running the simulation again on the new, larger network could produce the
exact same transmission tree as before. This is an extreme example, but it
serves to demonstrate the point that $N$ may be impossible to determine if the
epidemic does not come close to saturating the network.

% real data

We estimated the parameters of the \gls{BA} model for several published
datasets (see~\ref{tab:data}). \textcite{cuevas2009hiv} performed a
phylogenetic analysis of 261 patients of mixed risk groups in the Basque
Country, Spain, who were newly diagnosed between 2004 and 2007. Although their
data did not comprise a single cluster, they did find that almost half (47\%)
of the samples grouped in transmission clusters. \textcite{wang2015targeting}
reported on a cohort of acutely infected \gls{MSM} in Beijing, China. Through
phylogenetic analysis, they found that they had ``deeply sampled'' the local
\gls{MSM} network, suggesting that many of the samples could be
epidemiologically related. \textcite{novitsky2013phylogenetic} investigated a
local HIV epidemic in Mochudi, Botswana. They later extended their
analysis~\autocite{novitsky2014impact} to include an estimated 70\% of HIV
infected persons in the town. 

Our method has a number of caveats, perhaps the most significant being that it
takes a transmission tree as input. In reality, true transmission trees are not
available and must be estimated, often by way of a viral phylogeny. Although
this has been demonstrated to work in practice~\autocite{leitner1996accurate},
the topologies of a viral phylogeny and transmission tree can differ
significantly~\autocite{ypma2013relating}. Additionally, the \gls{BA} model
that we fitted here makes a number of unrealistic modelling assumptions, such
as that the network is connected and static, the nodes obey \gls{SI} dynamics,
and that the underlying behaviour of all nodes is the same. This last is
particularly problematic, as we showed by simulating a network where some nodes
exhibited a higher attachment power than others. The estimated attachment power
was simply the average of the two values, indicating that, although we could
characterize the network in aggregate, the estimated parameters could not be
said to apply to any individual node. Finally, the method is computationally
intensive, taking about a day when run on 20 cores in parallel.

\section*{Conclusions}

We developed a novel method which uses kernel-\gls{ABC} to fit contact network
models to transmission trees, and used it to fit the \gls{BA} preferential
attachment model to both simulated and real data. 

\setcounter{figure}{0}
\renewcommand{\thefigure}{S\arabic{figure}}

\printbibliography

\newpage

\section*{Supplemental Materials}

\begin{figure}[ht]
  \includegraphics[scale=0.8]{leventhal2012fig1}
  \caption{Reproduction of Figure 1A from \textcite{leventhal2012inferring}.}
  \label{fig:sf1}
\end{figure}

\begin{figure}
  \includegraphics{smc_test}
  \caption{Approximation of mixture of Gaussians used by
    \textcite{del2012adaptive} and \textcite{sisson2007sequential} to test
    \gls{SMC}. Solid black line indicates true distribution. Grey shaded area
    shows SMC approximation obtained with our implementation.}
  \label{fig:smctest}
\end{figure}

\end{document}
