Due to the rapid advancement of nucleotide sequencing technology, viral
sequence data data have become increasingly feasible to collect on a population
level. Through phylodynamic methods, these data offer a window into
epidemiological processes which would otherwise be virtually impossible to
study on a realistic scale. Although phylodynamic methods have been developed
to fit a wide variety of epidemiological models to phylogenetic
data~\autocite{volz2012complex, rasmussen2014phylodynamic}, our method is the
first (to our knowledge) which can fit models that do not assume panmixia. We
believe this capability will be of broad interest to the molecular evolution
and epidemiology community, as it widens the field of epidemiological
parameters which may be investigated through viral sequence data. In addition,
the characterization of local contact networks could be valuable from a public
health perspective, such as for investigating optimal vaccination strategies.
This information could assist in curtailing current epimemics, as well as
preventing future epidemics of different diseases over the same contact
network.

Contact network structure has a substantial impact on epidemic trajectory
\autocite{barthelemy2005dynamical, volz2007susceptible, volz2008sir}, and a few
methods have been developed to estimate network parameters from epidemiological
data \autocite{volz2007susceptible, britton2002bayesian,
groendyke2011bayesian}. It is known that contact network structure can have a
substantial impact on transmission tree shape \autocite{o2010contact,
robinson2013dynamics, leventhal2012inferring, colijn2014phylogenetic,
villandre2016assessment}.

Our work had three main aims. First, we developed phylodynamic a method to
estimate contact network model parameters. 


Our second aim was to investigate the parameters of the \gls{BA} model, in
terms of their impact on tree shape and our ability to reconstruct them with
kernel-\gls{ABC}. We determined through simulation studies that the
preferential attachment power $\alpha$ and number of infected nodes $I$ had a
substantial impact on transmission tree shape, and could be estimated using our
method.


The final aim of our work was to apply our method to real world
\gls{HIV} datasets. 

% ergm

